### PR TITLE
first draft

### PR DESCRIPTION
**Pull request by MelamudMichael**

### CODE REVIEW

File: include/uprotocol-cpp/proto/ultifi.cmake
Comment (stevenhartley) [inline include/uprotocol-cpp/proto/ultifi.cmake:1]: Please rename makefile, do not use this term in open source repo

File: include/uprotocol-cpp/transport/UTransport.h
Comment (stevenhartley) [inline include/uprotocol-cpp/transport/UTransport.h:1]: All uTransport APIs should be returning UStatus and not UCode

Comment (stevenhartley) [reply include/uprotocol-cpp/transport/UTransport.h:1]: Also we will be removing authenticate, it is not a client facing API needed to be defined in uTransport

File: include/uprotocol-cpp/transport/datamodel/UListener.h [hunk 1]
```diff
@@ -1,10 +1,9 @@
 #ifndef _ULISTENER_H_
 #define _ULISTENER_H_
 
-#include <include/uri/up_uri.h>
-#include <include/uri/up_uri_factory.h>
-#include <include/model/attributes.h>
-#include "UStatus.h"
+#include <uprotocol-cpp/uri/datamodel/UUri.h>
+#include <uprotocol-cpp/transport/datamodel/UAttributes.h>
```
Comment (stevenhartley) [inline include/uprotocol-cpp/transport/datamodel/UListener.h:5]: UUri, UAttributes, UMessage, UPayload are all defined in proto now, are you pulling them in?

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -1,3 +1,9 @@
+# Involve uFw-Ultifi build system; than exit
```
Comment (stevenhartley) [inline test/CMakeLists.txt:1]: Please remove ultifi references. What is this for?

File: test/factory/cloud_event_factory_test.cpp
Comment (stevenhartley) [inline test/factory/cloud_event_factory_test.cpp:1]: Why is this entire file commented out?

File: test/ultifi.cmake
Comment (stevenhartley) [inline test/ultifi.cmake:1]: Remove ultifi term.

File: test/uri/LongUriSerializerTest.cpp [hunk 3]
```diff
@@ -0,0 +1,790 @@
...
+using namespace cgreen;
+using namespace uprotocol::uri;
+
+#define assertTrue(a) assert_true(a)
+#define assertEquals(a, b) assert_true(b == a)
+#define assertFalse(a) assert_false(a)
+
+Describe(LongUriSerializer);
+
+BeforeEach(LongUriSerializer) {
```
Comment (stevenhartley) [inline test/uri/LongUriSerializerTest.cpp:38]: What are these for? BeforeEach() and AfterEach()?

**Review by stevenhartley — CHANGES_REQUESTED** [review]
Please rename the include folder uprotocol-cpp to uprotocol. Also do you plan to pull in the core-api data model?

**Comment by stevenhartley** [comment]
@MelamudMichael you also need to ensure all files including makefiles had the license headers like Java, rust, and python sdks.

File: include/uprotocol-cpp/proto/ultifi.cmake
Comment (MelamudMichael) [reply include/uprotocol-cpp/proto/ultifi.cmake:1]: @stevenhartley  i need to remove this file , it was added by Max, but later on we need to think how we going to add this files, as they are needed for internal builds

File: include/uprotocol-cpp/transport/datamodel/UListener.h (hunk 1 above)
Comment (MelamudMichael) [reply include/uprotocol-cpp/transport/datamodel/UListener.h:5]: @stevenhartley - Yes , on a later stage though 
Adi is working now on the UURi, 
Can we merge without them and do it in stages?

File: test/factory/cloud_event_factory_test.cpp
Comment (MelamudMichael) [reply test/factory/cloud_event_factory_test.cpp:1]: @stevenhartley  i will revert it

File: test/uri/LongUriSerializerTest.cpp (hunk 3 above)
Comment (MelamudMichael) [reply test/uri/LongUriSerializerTest.cpp:38]: this is part of CGREEN tests , we are in progress of changing to GTEST

File: include/uprotocol-cpp/proto/ultifi.cmake
Comment (stevenhartley) [reply include/uprotocol-cpp/proto/ultifi.cmake:1]: @MelamudMichael this is where a packaging and build tools like conan will come in handy, we can put the cross compilation rules in a separate "build profile"

File: include/uprotocol-cpp/transport/datamodel/UListener.h (hunk 1 above)
Comment (stevenhartley) [reply include/uprotocol-cpp/transport/datamodel/UListener.h:26]: yes

**Comment by stevenhartley** [comment]
Merging the branch, we will continue to work on c++ SDK, outstanding work items will be covered in the open issues

**Pull request merged**